bundles/org.openhab.binding.kaleidescape/src/main/java/org/openhab/binding/kaleidescape/internal/handler/KaleidescapeHandler.java

```diff
@@ -0,0 +1,603 @@
+            // case statement here
+            KaleidescapeMessageHandler.valueOf(evt.getKey()).handleMessage(evt.getValue(), this);
+
+            if (ThingStatusDetail.BRIDGE_OFFLINE.equals(thing.getStatusInfo().getStatusDetail())) {
+                // no longer in standby, update the status
+                updateStatus(ThingStatus.ONLINE, ThingStatusDetail.NONE, this.friendlyName);
+            }
+        } catch (IllegalArgumentException e) {
+            logger.debug("Unhandled message: key {} = {}", evt.getKey(), evt.getValue());
+        }
+    }
+
+    /**
+     * Schedule the reconnection job
+     */
+    private void scheduleReconnectJob() {
+        logger.debug("Schedule reconnect job");
+        cancelReconnectJob();
+        reconnectJob = scheduler.scheduleWithFixedDelay(() -> {
+            synchronized (sequenceLock) {
+                if (!connector.isConnected()) {
+                    logger.debug("Trying to reconnect...");
+                    closeConnection();
+                    String error = EMPTY;
+                    if (openConnection()) {
+                        try {
+                            Set<String> initialCommands = new HashSet<String>(Arrays.asList(GET_DEVICE_TYPE_NAME,
+                                    GET_FRIENDLY_NAME, GET_DEVICE_INFO, GET_SYSTEM_VERSION, GET_DEVICE_POWER_STATE,
+                                    GET_CINEMASCAPE_MASK, GET_CINEMASCAPE_MODE, GET_SCALE_MODE, GET_SCREEN_MASK,
+                                    GET_SCREEN_MASK2, GET_VIDEO_MODE, GET_UI_STATE, GET_HIGHLIGHTED_SELECTION,
+                                    GET_CHILD_MODE_STATE, GET_PLAY_STATUS, GET_MOVIE_LOCATION, GET_MOVIE_MEDIA_TYPE,
+                                    GET_PLAYING_TITLE_NAME));
+
+                            // Premiere Players and Cinema One support music
+                            if (PLAYER.equals(this.componentType) || CINEMA_ONE.equals(this.componentType)) {
+                                initialCommands.addAll(new ArrayList<String>(Arrays.asList(GET_MUSIC_NOW_PLAYING_STATUS,
+                                        GET_MUSIC_PLAY_STATUS, GET_MUSIC_TITLE)));
+                            }
+
+                            // everything after Premiere Player supports GET_SYSTEM_READINESS_STATE
+                            if (!PLAYER.equals(this.componentType)) {
+                                initialCommands.add(GET_SYSTEM_READINESS_STATE);
+                            }
+
+                            // only Strato supports the GET_*_COLOR commands
+                            if (STRATO.equals(this.componentType)) {
+                                initialCommands.addAll(
+                                        new ArrayList<String>(Arrays.asList(GET_VIDEO_COLOR, GET_CONTENT_COLOR)));
+                            }
+
+                            initialCommands.forEach(command -> {
```

Comment:
@fwolter It is possible that these are not needed since I have implemented the "REFRESH" actions for the channels? If the channels all refresh on reboot, they would cause all these same commands to be sent.

Reply:
They can be removed, then.

Reply:
@fwolter 
I am not sure. Is there a way to change the REFRESH behavior for the channels? I had seen refresh fire while developing the Nuvo binding. For a few channels, I was not handling the refresh type separately. Mistakenly I was doing command.toString() without checking the command type which ended up printing "REFRESH" on all my wall keypads :)

When I comment out the initial commands here, the channels do not seem to refresh on startup. I didn't want the same commands being sent by both reconnect and refresh. But I feel like it is better for reconnect to query the controller to get the current state of everything if the refresh behavior is inconsistent.

Reply:
I think it's ok to issue the queries twice on (re)connect for the sake of simple code.

Reply:
I have used `ExpiringCache` with a small timout, like 3 seconds. When the cache expires it will call the method passed. You could use this channels have the same command, such that when multiple of these channels are updated they are not all sending the command. And when using a cache you can probably iterate over the channels here, check if the channel is linked (i.e. if it actually make sense to update the channel) and then call `handleRefresh` with the channel instead of directly sending the command here. There is also an `ExpiringCacheMap` to handle multiple caches.